R/froll.R

```diff
@@ -111,6 +111,47 @@ make.roll.names = function(x.len, n.len, n, x.nm, n.nm, fun, adaptive) {
   ans
 }
 
+# irregularly spaced time series, helper for creating adaptive window size
+frolladapt = function(x, n, align="right", partial=FALSE, give.names=FALSE) {
+  x = unclass(x)
+  if (!is.numeric(x))
+    stopf("'x' must be of a numeric type")
+  if (!is.integer(x))
+    x = as.integer(x)
+  if (!is.numeric(n)) {
+    stopf("'n' must be an integer")
+  } else {
+    nms = names(n) ## only for give.names
+    if (!is.integer(n)) {
+      if (!fitsInInt32(n))
+        stopf("'n' must be an integer")
+      n = as.integer(n)
+    }
+  }
+  if (!length(n))
+    stopf("'n' must be non 0 length")
+  if (anyNA(n))
+    stopf("'n' must not have NAs")
+  if (!identical(align, "right"))
+    stopf("'align' other than 'right' has not yet been implemented")
+  if (!isTRUEorFALSE(partial))
+    stopf("'partial' must be TRUE or FALSE")
```

Comment:
(possible follow-up TODO):

```suggestion
    stopf("'%s' must be TRUE or FALSE", "partial")
```